src/AstSemanticChecker.cpp

```diff
@@ -433,6 +433,22 @@ void AstSemanticChecker::checkLiteral(
 
 void AstSemanticChecker::checkAggregator(
         ErrorReport& report, const AstProgram& program, const AstAggregator& aggregator) {
+
+    bool nestedFound = false;
+    // default value
+    auto* inner = &aggregator;
```

Comment:
Any reason the default value is `&aggregator`? Since it's tied to `nestedFound` maybe set it to `nullptr` and place an assertion right before using it later on (e.g. `assert(inner != nullptr && "inner aggregate cannot be nullptr")`), since it's an impossible case?

Reply:
yep ok i changed it

Reply:
oh. Ok

Reply:
I just base it off whether nullptr now so that null dereference is impossible

Reply:
Looks good!